### PR TITLE
Use mediaelementjs to get HLS playback in desktop browsers other than Safari

### PR DESCRIPTION
**Pull request by jancona**
Not sure if you'd be interested in this, but I thought I'd pass it along.

It still has some problems. Here's what I've noticed:
* Can't hide the player until the stream is ready. (See comments below.)
* Various issues in Chrome on my Android Lollipop (5.1) phone. Although it does play, it takes going in and out of full-screen mode and sometimes a reload of the to get it to work properly.

On the positive side, it seems to continue to work fine in desktop Safari and Chrome on my Macbook.

### CODE REVIEW

File: app/public/js/src/channel.coffee [hunk 1]
```diff
@@ -62,11 +62,14 @@ $ ->
 
     $(videoPlayer()).show()
```
Comment (jancona) [inline app/public/js/src/channel.coffee]: This doesn't work. I had to remove the "hide" on the video element in channel.erb.

File: app/public/js/src/channel.coffee [hunk 2]
```diff
@@ -62,11 +62,14 @@ $ ->
...
     videoPlayer().src = playUrl()
-    videoPlayer().play()
 
     if agentID && agentID[1] == 'iphone'
       window.location.href = playUrl()
 
+    $(videoPlayer()).mediaelementplayer(
+      success: (media, node, player) ->
+        player.play()
```
Comment (jancona) [inline app/public/js/src/channel.coffee]: For some reason the player doesn't actually start on Android. It does if you do it manually.

File: app/views/layout.erb [hunk 3]
```diff
@@ -28,8 +28,8 @@
     <script type="text/coffeescript" src="/js/src/settings.coffee"></script>
     <script src="/js/coffee-script.js"> </script>
 
-    <link href="http://vjs.zencdn.net/c/video-js.css" rel="stylesheet">
-    <script src="http://vjs.zencdn.net/c/video.js"></script>
+    <link  href="/mediaelement/mediaelementplayer.min.css" rel="stylesheet" type="text/css"/>
+    <script src="/mediaelement/mediaelement-and-player.min.js"></script>
```
Comment (jancona) [inline app/views/layout.erb]: Removing the videojs stuff didn't seem to have any effect, even on the old player. Perhaps it was left from an earlier experiment?

File: app/views/channel.erb [hunk 4]
```diff
@@ -2,5 +2,5 @@
   <div class="alert alert-block alert-info hide">
     <h4></h4>
   </div>
-  <video controls id="video-player" class="hide" src="/channels/<%= @channel %>.m3u8" width="640" height="360" data-tune-url="/channels/<%= @channel %>/tune" data-play-url="/channels/<%= @channel %>.m3u8" data-status-url="/channels/<%= @channel %>/status"></video>
+  <video controls id="video-player" class="xhide" src="/channels/<%= @channel %>.m3u8" width="640" height="360" data-tune-url="/channels/<%= @channel %>/tune" data-play-url="/channels/<%= @channel %>.m3u8" data-status-url="/channels/<%= @channel %>/status"></video>
```
Comment (jancona) [inline app/views/channel.erb]: See the comment on channel.coffee. If we can't get the show() to work, the class="xhide" should be removed.

**Comment by mmesnjak** [comment]
I would just want to confirm that it works in Chrome browser on Android 5.1.

**Comment by maddox** [comment]
Closing this in lieu of #55

**Pull request closed without merging**